csv.h

```diff
@@ -882,53 +883,53 @@ void parse(char *col, char *&x) {
 }
 
 template <class overflow_policy, class T>
-void parse_unsigned_integer(const char *col, T &x) {
-  x = 0;
+void parse_unsigned_integer(const char *col, optional<T> &x) {
+  x = {};
   while (*col != '\0') {
     if ('0' <= *col && *col <= '9') {
       T y = *col - '0';
-      if (x > (std::numeric_limits<T>::max() - y) / 10) {
-        overflow_policy::on_overflow(x);
+      if (x && *x > (std::numeric_limits<T>::max() - y) / 10) {
```

Comment:
Perhaps I'm missing something, but since we redefine `x = {}` above, isn't `x` always the null value at this line? Suggesting that the `if` condition will always be `false`.

Reply:
The first time through the `while` loop it'll be false, but subsequent iterations through the loop it might evaluate to `true`